src-ui.v3/src/Main.hs

```diff
@@ -731,11 +736,72 @@ pkgTagList m = Map.fromListWith (List.++) $ do
 joinE :: forall t m a. (Reflex t, MonadHold t m) => Event t (Event t a) -> m (Event t a)
 joinE = fmap switch . hold never
 
-button_ :: forall t m. (DomBuilder t m, PostBuild t m) => T.Text -> m (Event t ())
-button_ t = do
+clickElement_ :: forall t m. (DomBuilder t m, PostBuild t m) => T.Text -> T.Text -> m (Event t ())
+clickElement_ elm t = do
   let cfg = (def :: ElementConfig EventResult t (DomBuilderSpace m))
         & elementConfig_eventSpec %~ addEventSpecFlags (Proxy :: Proxy (DomBuilderSpace m)) Click (\_ -> preventDefault)
-  (e, _) <- element "button" cfg $ text t
+  (e, _) <- element elm cfg $ text t
   pure $ domEvent Click e
 
+stripSearch :: JSS.JSString -> JSS.JSString
+stripSearch sJ
+  | Just sJ'  <- JSS.stripPrefix "^" sJ = sJ'
+  | Just sJ' <- JSS.stripSuffix "$" sJ  = sJ'
+  | otherwise                           = sJ
 
+filterBySearchBox :: [JSS.JSString] -> JSS.JSString -> [JSS.JSString]
+filterBySearchBox pkgs sJ
+  | JSS.isInfixOf "^" sJ = filter (JSS.isPrefixOf (stripSearch sJ)) pkgs
+  | JSS.isInfixOf "$" sJ = filter (JSS.isSuffixOf (stripSearch sJ)) pkgs
+  | otherwise            = filter (JSS.isInfixOf (stripSearch sJ))pkgs
+
+splitInfixPkg :: JSS.JSString -> JSS.JSString -> (T.Text, T.Text, T.Text)
+splitInfixPkg stripSJ pkg = (frontT, midT, backT)
+  where
+    textS               = JSS.textFromJSString stripSJ
+    (frontT, reminderT) = T.breakOn textS (JSS.textFromJSString pkg)
+    (midT, backT)       = T.breakOnEnd textS reminderT
+
+calcMatchesNew :: [JSS.JSString] -> JSS.JSString -> Matches
+calcMatchesNew pkgs sJss = 
+  if JSS.length sJss < 3 
+  then matchesEmpty
+  else Matches { matchesInput = textS, matchesExact = exactMap, matchesInfix = othersMap}
+  where
+    textS                = JSS.textFromJSString sJss
+    stripSJ              = stripSearch sJss
+    (exactPkg, infixPkg) = List.partition (== stripSJ) pkgs
+    exactMap             = (Map.fromList . fmap (\p -> (JSS.textFromJSString p,()))) exactPkg
+    othersMap            = (Map.fromList . fmap (\p -> (JSS.textFromJSString p, splitInfixPkg stripSJ p))) (filterBySearchBox infixPkg sJss)
+
+searchBoxWidget :: forall t m. (SupportsServantReflex t m, MonadFix m, MonadIO m, MonadHold t m, PostBuild t m, DomBuilder t m, Adjustable t m, DomBuilderSpace m ~ GhcjsDomSpace)
+                => Dynamic t (Vector PkgN) 
+                -> m ()  
+searchBoxWidget dynPkgs0 = mdo
+  searchInputE <- elAttr "div" ("class" =: "item search right clearfix") $ do
+    divClass "text" $ text "Package Search"
+    sVal0 <- inputElement $ def & inputElementConfig_elementConfig . elementConfig_initialAttributes .~ fold ["class" =: "input-search","placeholder" =: "search..."]
+                                & inputElementConfig_setValue .~ clickPkgE
+    debounce 0.1 $ leftmost [clickPkgE, (_inputElement_input sVal0)]
```

Comment:
Does changing this to 0.2 or 0.3 make things smoother for you?

Reply:
0.1 is way smoother when we have long text such as "facebook", I think i'm going to stick with 0.1